ARSIS-Unity/Assets/Scripts/EventManager/EventTypes/BaseArsisEvent.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+using System.Diagnostics;
+using System.Collections.Generic;
+using System.Reflection;
+using UnityEngine;
+
+namespace ARSIS.EventManager
+{
+    /// <summary>
+    /// This is an interface that is used to ensure a level of
+    /// uniformity with what events can do.
+    /// </summary>
+    [System.Serializable]
+    public class BaseArsisEvent
+    {
+        public static readonly Dictionary<string, Type> EventTypes = new() {
+            { "telemetry", typeof(Telemetry) },
+        };
```

Comment:
What would you think about putting somthing like this in the `EventManager` to autogenerate the types?
```csharp
...
// stores reflection generated types
private Dictionary<string, Type> eventDictionary = new Dictionary<string, Type>();
```
```csharp
private struct TmpFormat
{
    public string type;
    public string data;
}

private static string EncodeEvent(BaseArsisEvent eventToEncode)
{
    string name = eventToEncode.GetType().Name;
    string body = JsonUtility.ToJson(eventToEncode);
    return JsonUtility.ToJson(new TmpFormat { type = name, data = body });
}

private BaseArsisEvent DecodeEvent(string json)
{
    TmpFormat tmp = JsonUtility.FromJson<TmpFormat>(json);
    Type type = eventDictionary[tmp.type];
    return (BaseArsisEvent)JsonUtility.FromJson(tmp.data, type);
}

```
```csharp

void Start() {
    ...
    // initalize the dict
    Type target = typeof(BaseArsisEvent);
    Assembly assembly = Assembly.GetAssembly(target);

    var types = assembly.GetTypes().Where(t => t.IsSubclassOf(target));

    foreach (var type in types)
    {
        eventDictionary.Add(type.Name, type);
    }
    ...
}

```

Reply:
nvm, I think I missed your `DoReflection` method